### PR TITLE
Implementation of ES6 syntax modules

### PR DESCRIPTION
**Pull request by MarkoKermi**
- Re-creation of  Awesome books project using ES6 syntax. 
- Re-creation of clock/date part using library Luxon.

### CODE REVIEW

File: modules/displayBook.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    const eachBook = {
+      title: bookTitle,
+      author: authorName,
+    };
+    this.bookArray.push(eachBook);
+  }
+
+  booksFilter(eachBook) {
+    this.bookArray = this.bookArray.filter((book) => book !== eachBook);
+  }
```
Comment (pinkmoon25) [inline modules/displayBook.js]: - [ ] good job refactoring :clap: , Please notice that one of the main goal of this project is to refactor it to use ES6 suntax, which uses arrow functions instead, I suggest we change all the functions to be arraw functions :pray: 
ex: 
```
const booksFilter = (eachBook) => {
    this.bookArray = this.bookArray.filter((book) => book !== eachBook);
  }
```

File: modules/script.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import Books from './displayBook.js';
+import { DateTime } from './luxon/src/luxon.js';
+
+const allBooks = new Books([]);
+
+function addToLocalStorage() {
+  const stringifyArray = JSON.stringify(allBooks.bookArray);
+  localStorage.setItem('storedBooks', stringifyArray);
+}
```
Comment (pinkmoon25) [inline modules/script.js]: - [ ] good job refactoring :clap: , Please notice that one of the main goals of this project is to refactor it to use ES6 syntax, which uses arrow functions instead, I suggest we change all the functions to be arrow functions :pray:

File: modules/script.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    removeButton.classList.add('remove');
+    removeButton.textContent = 'Remove';
+
+    removeButton.onclick = () => {
+      allBooks.booksFilter(allBooks.bookArray[i]);
+      addToLocalStorage();
+      displayBook();
+    };
+    container02.appendChild(removeButton);
+  }
```
Comment (pinkmoon25) [inline modules/script.js]: - [ ] good job refactoring :clap: , Please notice that one of the main goals of this project is to refactor it to use ES6 syntax, which uses arrow functions instead, I suggest we change all the functions to be arrow functions :pray:

File: modules/script.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+function getFromLocalStorage() {
+  const stringifyArray = localStorage.getItem('storedBooks');
+  allBooks.bookArray = JSON.parse(stringifyArray);
+  displayBook();
+}
```
Comment (pinkmoon25) [inline modules/script.js]: - [ ] good job refactoring :clap: , Please notice that one of the main goals of this project is to refactor it to use ES6 syntax, which uses arrow functions instead, I suggest we change all the functions to be arrow functions :pray:

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @MarkoKermi , 

Good job so far! you have done really well :clap:
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
linter checks :heavy_check_mark: 
github flow :heavy_check_mark: 
Add and remove functionality :heavy_check_mark: 
luxon :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

**- [ ] As mentioned in the project requirements we have to create the entry point for our JavaScript code called index.js or script.js if you prefer that, but the main thing is to have a single entry file and Import all your other js files which are suppose to be in the modules directory into this main entry file, Currently all your js file is in modules directory. can we try to fix this.**

**- [ ]  I noticed you have committed an empty js file clock.js, Please can we remove it as it is not the best practice to commit empty files to codebase, it can cause confusion :confused:** 

**- [ ] I see the code you have in your script.js is actually the displayBook code and it would be better if you name that displaybook file and your displayBoook file contains the structure of your book object i.e. your book class so you can rename it to just book.js. this will in turn make more sense and make the codebase follow best practices and ultimately will be more readable and easy to bugfix :bug:**  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @MarkoKermi,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Well organized project ✔️


appr
# Status: Approved ✔️✔️✔️
### Optional 
- Please for your next pull request remember to add a title. Also commit messages should be capitalized and in imperative form. 

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**